solstice/src/main/java/dev/equo/solstice/p2/P2Model.java

```diff
@@ -114,6 +115,24 @@ public P2Query query(P2Client.Caching caching) throws Exception {
 		return query;
 	}
 
+	public P2QueryResult queryUsingCache(P2Client.Caching cachingPolicy, boolean forceRecalculate) {
```

Comment:
I asked for `boolean forceRecalculate`, but I think I made a mistake. We should introduce `enum QueryCache { ALLOW, FORCE_RECALCULATE }`, I would put it in `dev.equo.solstice.p2`, and use that instead of the `boolean`.

This creates a name collision with the existing `QueryCache`. I would rename the existing `QueryCache` to `QueryCacheOnDisk` and make it package-private, it's an implementation detail that user's don't need to know about.